File: input.go
package main

import (
	"encoding/json"
	"errors"
	"fmt"
	"io"
	"io/ioutil"
	"net/http"
	"os"
)

func main() {
	if len(os.Args) <= 1 {
		fmt.Println("No day number was passed in")
		os.Exit(1)
	}

	cookie, err := getCookie()
	if err != nil {
		fmt.Printf("%s\n", err)
		os.Exit(1)
	}

	req, err := http.NewRequest(
		"GET",
		fmt.Sprintf("https://adventofcode.com/2019/day/%s/input",
			os.Args[1]),
		nil)
	if err != nil {
		fmt.Println("error: unable to form request")
		os.Exit(1)
	}

	req.AddCookie(
		&http.Cookie{
			Name:  "session",
			Value: cookie.Session})
	client := new(http.Client)
	resp, err := client.Do(req)
	if resp.StatusCode != 200 {
		fmt.Printf("error: status code %d\n", resp.StatusCode)
		os.Exit(1)
	}
	defer resp.Body.Close()

	io.Copy(os.Stdout, resp.Body)
}

type cookie struct {
	Session string `json:"session"`
}

func getCookie() (*cookie, error) {
	in, err := ioutil.ReadFile("cookie.json")
	if err != nil {
		return nil, errors.New("getCookie: error reading file")
	}

	c := new(cookie)
	err = json.Unmarshal(in, c)
	if err != nil {
		return nil, errors.New("getCookie: error decoding json")
	}
	return c, nil
}

File: input.py
#!/usr/bin/env python3

import json
import sys
import urllib
import urllib.request

def main(*args, **kwargs):
    with open("cookie.json") as f:
        cookie = json.load(f)
        req = urllib.request.Request(
            "https://adventofcode.com/{}/day/{}/input".format(kwargs['year'], kwargs['day'])
        )
        req.add_header("Cookie", "session={}".format(cookie['session']))
        with urllib.request.urlopen(req) as response:
            print(response.read().decode("utf-8").strip())

if __name__ == "__main__":
    if len(sys.argv) < 2:
        print("Usage: ./input.py <year> <day>")
        sys.exit(1)

    main(year=sys.argv[1], day=sys.argv[2])
